lib/daru/dataframe.rb

```diff
@@ -423,17 +423,17 @@ def row
     end
 
     # Duplicate the DataFrame entirely.
-    # 
+    #
     # == Arguments
-    # 
-    # * +vectors_to_dup+ - An Array specifying the names of Vectors to 
+    #
+    # * +vectors_to_dup+ - An Array specifying the names of Vectors to
     # be duplicated. Will duplicate the entire DataFrame if not specified.
     def dup vectors_to_dup=nil
       vectors_to_dup = @vectors.to_a unless vectors_to_dup
 
       src = []
       vectors_to_dup.each do |vec|
-        src << @data[@vectors[vec]].to_a
+        src << @data[@vectors[vec]].dup.to_a
```

Comment:
If you do `vector.dup.to_a` you will first duplicate a `Daru::Vector` and convert it to an Array. A faster and smarter  way to do this would be `vector.to_a.dup` so that only a Ruby Array is being duplicated.

Please make this change before I can merge.